src/ngraph/op/softmax.cpp

```diff
@@ -28,25 +28,32 @@
 using namespace std;
 using namespace ngraph;
 
+// *** SOFTMAX OP SET 0 ***
 const string op::Softmax::type_name{"Softmax"};
 
-op::Softmax::Softmax(const Output<Node>& arg, const AxisSet& axes)
-    : UnaryElementwiseArithmetic(arg)
+op::v0::Softmax::Softmax(const Output<Node>& arg, const AxisSet& axes)
+    : Op({arg})
     , m_axes(axes)
 {
     constructor_validate_and_infer_types();
 
+    const PartialShape& input_shape = get_input_partial_shape(0);
     for (auto axis : m_axes)
     {
         NODE_VALIDATION_CHECK(this,
-                              axis < get_shape().size(),
+                              axis < static_cast<size_t>(input_shape.rank()),
```

Comment:
what about if rank is dynamic ?

Reply:
As I know `rank` method supports `dynamic` case

Reply:
If rank would be dynamic then static_cast will throw an exception.

Reply:
@tsocha Thanks, you are right, I added `NODE_VALIDATION_CHECK` to provide it